test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -27,6 +27,25 @@ def test_configure_use_grpc
     assert_true d.instance.instance_variable_get(:@use_grpc)
   end
 
+  def test_user_agent
+    setup_gce_metadata_stubs
+
+    user_agent = nil
+    # Record user agent when creating a GRPC::Core::Channel.
+    GRPC::Core::Channel.class_eval do
+      old_initialize = instance_method(:initialize)
+      define_method(:initialize) do |url, args, creds|
+        user_agent = args['grpc.primary_user_agent']
+        old_initialize.bind(self).call(url, args, creds)
+      end
+    end
+
+    d = create_driver
+    d.instance.send :init_api_client
+    assert_match Regexp.new(Fluent::GoogleCloudOutput::PLUGIN_NAME.to_s), \
```

Comment:
Should we test against the full version?
"#{PLUGIN_NAME}/#{PLUGIN_VERSION} grpc-ruby/#{GRPC::VERSION} #{Google::Apis::OS_VERSION}"

Reply:
I think the key reason we set the User Agent is to be able to identify the requests coming from the logging agent. So this tests the minimal such identification. I think a more precise test will be more fragile and won't buy us anything — it'll be like a change detector test.

Reply:
We also want to guarantee at least the plugin version is set (not empty) and correct. The test can check it against the variable in the main class so it's not a moving target.

Reply:
Done.